src/opm/parser/eclipse/EclipseState/Schedule/Group/Group.cpp

```diff
@@ -343,51 +384,47 @@ Group::ProductionControls Group::productionControls(const SummaryState& st) cons
     return pc;
 }
 
-Group::InjectionControls Group::injectionControls(const SummaryState& st) const {
+Group::InjectionControls Group::injectionControls(Phase phase, const SummaryState& st) const {
     Group::InjectionControls ic;
+    const auto& inj = this->injection_properties.at(phase);
+
+    ic.phase = inj.phase;
+    ic.cmode = inj.cmode;
+    ic.surface_max_rate = UDA::eval_group_uda_rate(inj.surface_max_rate, this->m_name, st, this->udq_undefined, ic.phase, this->unit_system);
```

Comment:
injection_control is not copied over to the injectionControls 
but later used when calling 

bool Group::InjectionControls::has_control(InjectionCMode cmode_arg) const {
    return detail::has_control(this->injection_controls, cmode_arg);
}